pcbdraw/pcbdraw.py

```diff
@@ -581,13 +605,22 @@ def component_to_board_scale(svg):
     return width / vw, height / vh
 
 def component_from_library(lib, name, value, ref, pos, usedComponents, comp,
-                           highlight, silent, no_warn_back):
+                           highlight, silent, no_warn_back, style, tht_resistor_settings):
+
     if not name:
         return
     if comp["filter"] is not None and ref not in comp["filter"]:
         return
 
-    unique_name = f"{lib}__{name}"
+    # If the part is a THT resistor, change it's value if the parameter custom_res_color has
+    # the resistor specifing a new value
+    if lib == "Resistor_THT":
+        if tht_resistor_settings is not None:
+            if ref in tht_resistor_settings:
+                if 'override_val' in tht_resistor_settings[ref]:
+                    value = tht_resistor_settings[ref]['override_val']
```

Comment:
I would prefere to not check for the library as the users might use custom libraries. We should only check if there is a resistor value specified; if so we try to color it. If it fails we notify the user and stop.

Reply:
I think instead of finding a resistor value, as any given number for the value can be one, it's better to check if the component supports the color bands as per my recent change:
`if componentElement.find(".//*[@id='{}res_band1']".format(svg_prefix)) is not None:`

Reply:
I like to solution